### ISSUE TITLE
"node_sqlite3.node is not a valid Win32 application" on Windows on ARM64

### ISSUE THREAD

**Issue by ktos** [continuedev/continue#839]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Windows 11 [Version 10.0.22631.3085], ARM64
- Continue: v0.8.11 (but the same problem is in v0.9.58)
- IDE: VSCode 1.86.1
```

### Description

After just installing the current version of continue (and pre-release, the same problem) and opening its window, the "clock" icon stays forever and nothing more happens, but I finally managed to find the appropriate VSCode log section where there is an error message:

```
2024-02-09 23:31:33.438 [error] Error: \\?\c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\build\Release\node_sqlite3.node is not a valid Win32 application.
\\?\c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\build\Release\node_sqlite3.node
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:2:2131)
    at Object.<anonymous> (node:internal/modules/cjs/loader:1403:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:2:2131)
    at Module.load (node:internal/modules/cjs/loader:1173:32)
    at node:internal/modules/cjs/loader:1014:12
    at Function.<anonymous> (node:electron/js2c/asar_bundle:2:13327)
    at Function.l._load (c:\Users\ktos\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:173:5635)
    at Function.r._load (c:\Users\ktos\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:170:29791)
    at Function.t._load (c:\Users\ktos\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:135:35292)
    at Module.require (node:internal/modules/cjs/loader:1197:19)
    at h (c:\Users\ktos\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\loader.js:4:647)
    at bindings (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:45868:52)
    at ../../core/node_modules/sqlite3/lib/sqlite3-binding.js (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:45940:41)
    at __require (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:13:51)
    at ../../core/node_modules/sqlite3/lib/sqlite3.js (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:45990:20)
    at __require (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:13:51)
    at ../../core/indexing/docs/db.ts (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:93109:30)
    at __init (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:10:56)
    at ../../core/indexing/docs/index.ts (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:99781:5)
    at __init (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:10:56)
    at src/debugPanel.ts (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:163096:5)
    at __init (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:10:56)
    at src/commands.ts (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:163482:5)
    at __init (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:10:56)
    at src/activation/activate.ts (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:171636:5)
    at __init (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:10:56)
    at c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:171683:89
    at dynamicImportAndActivate (c:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\extension.js:171683:53)
```

The file "node_sqlite3.node" referenced there is actually a x86-64 DLL, not an ARM64 DLL, so I believe the problem may lay in there:

```
C:\Users\ktos\.vscode\extensions\continue.continue-0.8.11-win32-arm64\out\build\Release>file node_sqlite3.node
node_sqlite3.node: PE32+ executable (DLL) (GUI) x86-64, for MS Windows, 7 sections
```

A bit related to the #724 where I asked for Win32-ARM64 support related to the esbuild - the pre-release version v0.9.8 was behaving very similarly but I couldn't pinpoint the actual error message until today.

**Comment by DanielYzd** [comment]
How can I solve the same problem I encountered?

**Comment by q479902964** [comment]
the same problem +1

**Comment by permpkin** [comment]
Looks to be an issue when building sqlite3 and onnxruntime-node, Same issue occurs for me on Apple silicon, Solution for me was to run the following ( the same approach should also work on W11 ARM as well )...
```shell
# change to plugin directory
cd ~/.vscode/extensions/continue.continue-*

# remove sqlite3 and onnxruntime libraries
npm uninstall sqlite3 onnxruntime-node

# reinstall with targetet architecture flags 
npm install sqlite3 onnxruntime-node --build-from-source --target_arch=arm64 --fallback-to-build

# copy onnxruntime binaries to output folders
cp ./node_modules/libonnxruntime.1.17.3.dylib ./bin/napi-v3/darwin/arm64/libonnxruntime.1.17.3.dylib
cp ./node_modules/onnxruntime_binding.node ./bin/napi-v3/darwin/arm64/onnxruntime_binding.node

# copy sqlite binary to build output folder
cp ./node_modules/sqlite3/build/Release/node_sqlite3.node ./out/build/Release/node_sqlite3.node
```

**Comment by dschonholtz** [comment]
Relevant stack overflow post: https://stackoverflow.com/questions/72553650/how-to-get-node-sqlite3-working-on-mac-m1

**Comment by theNailz** [comment]
Same error for **Win11 x64**.

The file `C:\Users\me\AppData\Roaming\JetBrains\GoLand2024.1\plugins\continue-intellij-extension\core\build\Release\node_sqlite3.node` is the arm32 version. I 'fixed' it (this error, but processing seems to hang) by copying the right binary from `C:\Users\me\AppData\Roaming\JetBrains\GoLand2024.1\plugins\continue-intellij-extension\core\win32-x64`.

Now it's stuck at "Completed indexing chunks", with no errors in the core.log. Hope this might give anyone a clue or point of reference.

I'm bummed that my first experience with Continue is a bad experience. It seemed promising.

**Comment by sestinj** [comment]
@theNailz sorry to hear about this. Luckily I think it's related to a fix I'll be making today—were you on Continue version 0.0.54?

**Comment by theNailz** [comment]
> @theNailz sorry to hear about this. Luckily I think it's related to a fix I'll be making today—were you on Continue version 0.0.54?

Yes 0.0.54

**Comment by coinzz** [comment]
Same error for me after upgrading from W10 Jetbrains `0.0.56` to `0.0.59`. Got fixed by manually copying the correct compiled file as already explained from `core\win32-x64\node_sqlite3.node` to `build\Release` folder

**Comment by wenwen12345** [comment]
I get the same BUG in Webstorm windows11 x64,is anyone know how to fix?

**Comment by bakermat** [comment]
I'm also encountering this issue, on a brand new Windows 11 ARM64, using VS Code (ARM). Tried the latest release *0.8.46) as well as the latest pre-release, to no avail.

From the debug console:
`Activating extension 'Continue.continue' failed: \\?\c:\Users\x\.vscode\extensions\continue.continue-0.8.46-win32-arm64\out\build\Release\node_sqlite3.node is not a valid Win32 application.
\\?\c:\Users\x\.vscode\extensions\continue.continue-0.8.46-win32-arm64\out\build\Release\node_sqlite3.node.`

I tried the suggestions above from https://github.com/continuedev/continue/issues/839#issuecomment-2081875150 and https://github.com/continuedev/continue/issues/839#issuecomment-2215973239 but the error message stays the same, even after disabling / enabling the extension as well as restarting VS Code.

**Comment by curtfehr** [comment]
Bump, still occurring Windows 11 ARM64 (Snapdragon Elite)

**Comment by KrFeher** [comment]
Same problem here with snapdragone Elite. Anyone has a solution?

**Comment by danielme85** [comment]
I was also curios about the ARM64 Windows experience so I went and got a hold of one of the new Snapdragon Elite laptops. It's been working out great so far except for this plugin. I just started using a local ollama setup for code completion and this plugin works great for x64 on windows and the ARM M1 apple, not so much for winARM. 

I know its probably a small user base right now, but man: the snapdragon elite laptops are awesome: fast, (relatively) cheap vs Apple and all day battery life, so I bet ya it will have more users soon. 

Long story short, +1 request for a winARM64 build/fix please!

**Comment by robvoi** [comment]
+1 here too
Would love to use it on my ARM Windows machine.

**Comment by svantesorberg** [comment]
I managed to run the VSCode extension on my Snapdragon X Elite laptop after building `sqlite3` (as described higher up in this thread) and `lancedb` locally with some tinkering.

I'd be happy to share binaries or walk you through the build process. Though I guess the only sustainable way forward is for the developers of these upstream packages to provide pre-built binaries for windows arm64?

* For `sqlite3` one way forward could be switching to `better-sqlite3` as proposed [here](https://github.com/continuedev/continue/issues/1207).
* For `lancedb` I don't imagine it's viable to switch it out for something else, but maybe we can get them to provide binaries for windows arm64. The build was straightforward once I managed to straighten out the build dependencies, at least on an Windows arm64 platform, no idea about cross-compiling.

**Comment by svantesorberg** [comment]
> > I managed to run the VSCode extension on my Snapdragon X Elite laptop after building `sqlite3` (as described higher up in this thread) and `lancedb` locally with some tinkering.
> > I'd be happy to share binaries or walk you through the build process. Though I guess the only sustainable way forward is for the developers of these upstream packages to provide pre-built binaries for windows arm64?
> > 
> > * For `sqlite3` one way forward could be switching to `better-sqlite3` as proposed [here](https://github.com/continuedev/continue/issues/1207).
> > * For `lancedb` I don't imagine it's viable to switch it out for something else, but maybe we can get them to provide binaries for windows arm64. The build was straightforward once I managed to straighten out the build dependencies, at least on an windows arm64 platform, no idea about cross-compiling.
> 
> I spent half a day trying to rebuild this damn module on Windows ARM64, but nothing worked. Have you managed to get Continue working on Windows 11 ARM64? Could you share the necessary files?

Here you go, hope this is permitted.

From the continue folder (e.g. `C:\Users\<name>\.vscode\extensions\continue.continue-x.y.z-win32-arm64` ), you need to put `node_sqlite3.node` under `.\out\build\Release\` and the `vectordb-win32-arm64-msvc` folder under `.\out\node_modules\@lancedb\`    
[bundles.zip](https://github.com/user-attachments/files/16968419/bundles.zip)

Keep in mind that I just built this off lancedb master, so no guarantees that it will work for updated versions of `continue` (or the current one, for that matter).

I'll see if I can cross-compile `lancedb` for arm64 from my windows x86 machine. If that works I guess there's a chance they could integrate it in their normal builds and provide binaries.

**Comment by antonovkz** [comment]
> > > I managed to run the VSCode extension on my Snapdragon X Elite laptop after building `sqlite3` (as described higher up in this thread) and `lancedb` locally with some tinkering.
> > > I'd be happy to share binaries or walk you through the build process. Though I guess the only sustainable way forward is for the developers of these upstream packages to provide pre-built binaries for windows arm64?
> > > 
> > > * For `sqlite3` one way forward could be switching to `better-sqlite3` as proposed [here](https://github.com/continuedev/continue/issues/1207).
> > > * For `lancedb` I don't imagine it's viable to switch it out for something else, but maybe we can get them to provide binaries for windows arm64. The build was straightforward once I managed to straighten out the build dependencies, at least on an windows arm64 platform, no idea about cross-compiling.
> > 
> > 
> > I spent half a day trying to rebuild this damn module on Windows ARM64, but nothing worked. Have you managed to get Continue working on Windows 11 ARM64? Could you share the necessary files?
> 
> Here you go, hope this is permitted. [bundles.zip](https://github.com/user-attachments/files/16968419/bundles.zip)
> 
> From the continue folder (e.g. `C:\Users\<name>\.vscode\extensions\continue.continue-x.y.z-win32-arm64` ), you need to put `node_sqlite3.node` under `.\out\build\Release\` and the `vectordb-win32-arm64-msvc` folder under `.\out\node_modules\@lancedb\`
> 
> Keep in mind that I just built this off lancedb master, so no guarantees that it will work for updated versions of `continue` (or the current one, for that matter).
> 
> I'll see if I can cross-compile `lancedb` for arm64 from my windows x86 machine. If that works I guess there's a chance they could integrate it in their normal builds and provide binaries.

This worked! You saved me, I was so desperate that I installed the x64 version of VSCode in emulation mode, and of course, that's not ideal. Now I’ve disabled automatic updates and will manually check for new versions from time to time, both to test compatibility with your files and in the hope that the Continue extension developers will provide a fix. Thank you.

**Comment by ruverav** [comment]
@svantesorberg I wanted to try Continue in my ARM64 window laptop and I couldn't. Thank you, it's working fine.

**Comment by sestinj** [comment]
Thanks all for providing some workarounds here. I'm going to reach out to LanceDB and SQLite to see if they might be willing to include windows arm64 builds in their releases, but will also take a stab at doing the build myself soon (now that it looks finally available in GH actions! https://github.blog/news-insights/product-news/arm64-on-github-actions-powering-faster-more-efficient-build-systems/)

**Comment by bold84** [comment]
I created a script that simplifies the update with the files provided by @svantesorberg .
Just download the bundles.zip and place it into the same directory as this script. 
Executing it should update the relevant files of the current continue extension so the extension should work on Windows on ARM64. 


```
import os
import shutil
import zipfile

def get_extension_directory(username):
    base_path = f"C:\\Users\\{username}\\.vscode\\extensions"
    all_dirs = [d for d in os.listdir(base_path) if os.path.isdir(os.path.join(base_path, d))]
    
    matching_dirs = [d for d in all_dirs if d.startswith("continue.continue-") and "win32-arm64" in d]
    
    if len(matching_dirs) == 0:
        print(f"No directories found matching the criteria.")
        # List all directories in the base path for manual verification
        print(f"All directories in {base_path}:")
        for dir_name in all_dirs:
            print(dir_name)
        raise Exception("Could not find any matching extension directory.")
    elif len(matching_dirs) > 1:
        print(f"Found multiple directories matching the criteria: {matching_dirs}")
        raise Exception("Found more than one matching extension directory.")
    
    return os.path.join(base_path, matching_dirs[0])

def unpack_bundle(zip_path, extract_to):
    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
        zip_ref.extractall(extract_to)

def main():
    # Get the current user's username
    try:
        username = os.getlogin()
    except OSError:
        print("Could not determine the current user. Using environment variable USERPROFILE.")
        username = os.path.basename(os.environ['USERPROFILE'])
    
    print(f"Using username: {username}")
    
    # Locate the VS Code extension directory
    try:
        extension_dir = get_extension_directory(username)
        print(f"Found extension directory: {extension_dir}")
    except Exception as e:
        print(e)
        return
    
    # Define paths for unpacking and copying
    script_dir = os.path.dirname(os.path.abspath(__file__))
    bundle_zip_path = os.path.join(script_dir, "bundles.zip")  # Renamed to bundles.zip
    temp_extract_dir = os.path.join(script_dir, "temp_extract")
    
    if not os.path.exists(bundle_zip_path):
        raise Exception("bundles.zip not found in the script directory.")
    
    # Unpack bundles.zip
    unpack_bundle(bundle_zip_path, temp_extract_dir)
    print(f"Unpacked {bundle_zip_path} to {temp_extract_dir}")
    
    # Define source and destination paths for copying files
    node_sqlite3_source = os.path.join(temp_extract_dir, "node_sqlite3.node")
    vectordb_source = os.path.join(temp_extract_dir, "vectordb-win32-arm64-msvc")
    
    out_build_release_dest = os.path.join(extension_dir, "out", "build", "Release")
    node_modules_lancedb_dest = os.path.join(extension_dir, "out", "node_modules", "@lancedb", "vectordb-win32-arm64-msvc")  # Ensure the directory is copied as a whole
    
    # Ensure destination directories exist
    os.makedirs(out_build_release_dest, exist_ok=True)
    os.makedirs(os.path.dirname(node_modules_lancedb_dest), exist_ok=True)  # Create parent directories if they don't exist
    
    # Copy files and directories
    shutil.copy2(node_sqlite3_source, out_build_release_dest)
    print(f"Copied {node_sqlite3_source} to {out_build_release_dest}")
    
    shutil.rmtree(node_modules_lancedb_dest, ignore_errors=True)  # Remove existing directory if it exists
    shutil.copytree(vectordb_source, node_modules_lancedb_dest)
    print(f"Copied {vectordb_source} to {node_modules_lancedb_dest}")
    
    # Clean up temporary extraction directory
    shutil.rmtree(temp_extract_dir)
    print(f"Cleaned up temporary extraction directory: {temp_extract_dir}")

if __name__ == "__main__":
    main()
```

**Comment by mohdmerajmalik** [comment]
Hi, I'm also experiencing this issue on my Windows and mac system. I'm wondering if there has been any progress or resolution on this yet?

If not, could you confirm if the following workaround is the recommended approach?

**Comment by mehraj43** [comment]
Hi, I'm also experiencing this issue on my Windows and mac system. I'm wondering if there has been any progress or resolution on this yet?

If not, could you confirm if the following workaround is the recommended approach or is there some other approach?

**Comment by xgdgsc** [comment]
Thanks for the above bundle.zip and python script. It works. But sometimes the window used to work Continue Extension start to hang again while opening a new vscode window works. What could be the reason?

**Comment by antonovkz** [comment]
> I created a script that simplifies the update with the files provided by @svantesorberg . Just download the bundles.zip and place it into the same directory as this script. Running this script should update the relevant files of the current continue extension so the extension should work on Windows on ARM64.
> 
> ```python
> import os
> import shutil
> import zipfile
> 
> def get_extension_directory(username):
>     base_path = f"C:\\Users\\{username}\\.vscode\\extensions"
>     all_dirs = [d for d in os.listdir(base_path) if os.path.isdir(os.path.join(base_path, d))]
>     
>     matching_dirs = [d for d in all_dirs if d.startswith("continue.continue-") and "win32-arm64" in d]
>     
>     if len(matching_dirs) == 0:
>         print(f"No directories found matching the criteria.")
>         raise Exception("Could not find any matching extension directory.")
>     elif len(matching_dirs) > 1:
>         print(f"Found multiple directories matching the criteria: {matching_dirs}")
>         raise Exception("Found more than one matching extension directory.")
>     
>     return os.path.join(base_path, matching_dirs[0])
> 
> def unpack_bundle(zip_path, extract_to):
>     with zipfile.ZipFile(zip_path, 'r') as zip_ref:
>         zip_ref.extractall(extract_to)
> 
> def main():
>     # Get the current user's username
>     try:
>         username = os.getlogin()
>     except OSError:
>         print("Could not determine the current user. Using environment variable USERPROFILE.")
>         username = os.path.basename(os.environ['USERPROFILE'])
>     
>     print(f"Using username: {username}")
>     
>     # Locate the VS Code extension directory
>     try:
>         extension_dir = get_extension_directory(username)
>         print(f"Found extension directory: {extension_dir}")
>     except Exception as e:
>         print(e)
>         return
>     
>     # Define paths for unpacking and copying
>     script_dir = os.path.dirname(os.path.abspath(__file__))
>     bundle_zip_path = os.path.join(script_dir, "bundles.zip")  # Renamed to bundles.zip
>     temp_extract_dir = os.path.join(script_dir, "temp_extract")
>     
>     if not os.path.exists(bundle_zip_path):
>         raise Exception("bundles.zip not found in the script directory.")
>     
>     # Unpack bundles.zip
>     unpack_bundle(bundle_zip_path, temp_extract_dir)
>     print(f"Unpacked {bundle_zip_path} to {temp_extract_dir}")
>     
>     # Define source and destination paths for copying files
>     node_sqlite3_source = os.path.join(temp_extract_dir, "node_sqlite3.node")
>     vectordb_source = os.path.join(temp_extract_dir, "vectordb-win32-arm64-msvc")
>     
>     out_build_release_dest = os.path.join(extension_dir, "out", "build", "Release")
>     node_modules_lancedb_dest = os.path.join(extension_dir, "out", "node_modules", "@lancedb", "vectordb-win32-arm64-msvc")  # Ensure the directory is copied as a whole
>     
>     # Ensure destination directories exist
>     os.makedirs(out_build_release_dest, exist_ok=True)
>     os.makedirs(os.path.dirname(node_modules_lancedb_dest), exist_ok=True)  # Create parent directories if they don't exist
>     
>     # Copy files and directories
>     shutil.copy2(node_sqlite3_source, out_build_release_dest)
>     print(f"Copied {node_sqlite3_source} to {out_build_release_dest}")
>     
>     shutil.rmtree(node_modules_lancedb_dest, ignore_errors=True)  # Remove existing directory if it exists
>     shutil.copytree(vectordb_source, node_modules_lancedb_dest)
>     print(f"Copied {vectordb_source} to {node_modules_lancedb_dest}")
>     
>     # Clean up temporary extraction directory
>     shutil.rmtree(temp_extract_dir)
>     print(f"Cleaned up temporary extraction directory: {temp_extract_dir}")
> 
> if __name__ == "__main__":
>     main()
> ```

Here is the revised version of your PowerShell code, rewritten using a neural network ;), with an added check for the VSCodium fork installation:
```

# Get current user's username
$Username = $env:USERNAME
Write-Host "Using username: $Username"

# Define paths
$VscodeExtensionsPath = "C:\Users\$Username\.vscode\extensions"
$VscodeOssExtensionsPath = "C:\Users\$Username\.vscode-oss\extensions"

# Check if the directories exist
$VscodeExists = Test-Path $VscodeExtensionsPath
$VscodeOssExists = Test-Path $VscodeOssExtensionsPath

if ($VscodeExists -and -not $VscodeOssExists) {
    $ExtensionBasePath = $VscodeExtensionsPath
    Write-Host "Using extensions path: $ExtensionBasePath"
}
elseif (-not $VscodeExists -and $VscodeOssExists) {
    $ExtensionBasePath = $VscodeOssExtensionsPath
    Write-Host "Using extensions path: $ExtensionBasePath"
}
elseif ($VscodeExists -and $VscodeOssExists) {
    Write-Host "Both .vscode and .vscode-oss extensions directories exist."
    Write-Host "1) $VscodeExtensionsPath"
    Write-Host "2) $VscodeOssExtensionsPath"
    $choice = Read-Host "Enter 1 to use .vscode, 2 to use .vscode-oss"
    if ($choice -eq "1") {
        $ExtensionBasePath = $VscodeExtensionsPath
    } elseif ($choice -eq "2") {
        $ExtensionBasePath = $VscodeOssExtensionsPath
    } else {
        Write-Host "Invalid choice. Exiting."
        exit
    }
    Write-Host "Using extensions path: $ExtensionBasePath"
}
else {
    Write-Host "Neither .vscode nor .vscode-oss extensions directories exist. Exiting."
    exit
}

# Locate the VS Code extension directory
$AllDirs = Get-ChildItem -Path $ExtensionBasePath -Directory

$MatchingDirs = $AllDirs | Where-Object { $_.Name -like "continue.continue-*" -and $_.Name -like "*win32-arm64*" }

if ($MatchingDirs.Count -eq 0) {
    Write-Host "No directories found matching the criteria."
    exit
}
elseif ($MatchingDirs.Count -gt 1) {
    Write-Host "Found multiple directories matching the criteria:"
    $MatchingDirs | ForEach-Object { Write-Host $_.Name }
    Write-Host "Exiting."
    exit
}
else {
    $ExtensionDir = $MatchingDirs[0].FullName
    Write-Host "Found extension directory: $ExtensionDir"
}

# Define paths for unpacking and copying
$ScriptDir = Split-Path -Parent $MyInvocation.MyCommand.Definition
$BundleZipPath = Join-Path $ScriptDir "bundles.zip"
$TempExtractDir = Join-Path $ScriptDir "temp_extract"

if (-not (Test-Path $BundleZipPath)) {
    Write-Host "bundles.zip not found in the script directory."
    exit
}

# Unpack bundles.zip
Expand-Archive -LiteralPath $BundleZipPath -DestinationPath $TempExtractDir -Force
Write-Host "Unpacked $BundleZipPath to $TempExtractDir"

# Define source and destination paths for copying files
$NodeSqlite3Source = Join-Path $TempExtractDir "node_sqlite3.node"
$VectorDbSource = Join-Path $TempExtractDir "vectordb-win32-arm64-msvc"

$OutBuildReleaseDest = Join-Path $ExtensionDir "out\build\Release"
$NodeModulesLanceDbDest = Join-Path $ExtensionDir "out\node_modules\@lancedb\vectordb-win32-arm64-msvc"

# Ensure destination directories exist
if (-not (Test-Path $OutBuildReleaseDest)) {
    New-Item -ItemType Directory -Path $OutBuildReleaseDest -Force | Out-Null
}

$NodeModulesLanceDbDestParent = Split-Path $NodeModulesLanceDbDest
if (-not (Test-Path $NodeModulesLanceDbDestParent)) {
    New-Item -ItemType Directory -Path $NodeModulesLanceDbDestParent -Force | Out-Null
}

# Copy files and directories
Copy-Item -Path $NodeSqlite3Source -Destination $OutBuildReleaseDest -Force
Write-Host "Copied $NodeSqlite3Source to $OutBuildReleaseDest"

# Remove existing directory if it exists
if (Test-Path $NodeModulesLanceDbDest) {
    Remove-Item -Path $NodeModulesLanceDbDest -Recurse -Force
}

Copy-Item -Path $VectorDbSource -Destination $NodeModulesLanceDbDest -Recurse -Force
Write-Host "Copied $VectorDbSource to $NodeModulesLanceDbDest"

# Clean up temporary extraction directory
Remove-Item -Path $TempExtractDir -Recurse -Force
Write-Host "Cleaned up temporary extraction directory: $TempExtractDir"


```

**Comment by svantesorberg** [comment]
There is now some work going on in the [lancedb project](https://github.com/lancedb/lancedb/issues/1841) to get builds working for win32-arm64. Maybe you had something to do with this @sestinj :)

I'll try to help out if I can be of any use there, but I'm quite busy over the next few days so maybe else will work it out before I even get started.

**Comment by Amrsatrio** [comment]
Hi, same problem here on my X Elite machine running 26100.2314. Both the VSCode and JetBrains extensions do not work at all. I've configured Ollama with the recommended Qwen 2.5 1.5B model and it works fine. I haven't tried the solution above which is for VSCode only, and I don't really have time to try to compile the JetBrains extension for arm64 if that even works at the time of writing. Really looking forward for ARM64 support 🙏

**Comment by Volko61** [comment]
Thanks a lot for the scripts, they worked !!

**Comment by SpontaneousDuck** [comment]
lancedb now has an arm64 build as of v0.14: https://github.com/lancedb/lancedb/issues/1841

**Comment by Patrick-Erichsen** [comment]
Thanks for the heads up here @SpontaneousDuck , that is huge news! Going to circle back to this shortly to get that new build pulled in.

**Comment by Patrick-Erichsen** [comment]
We've got a preview build for win-arm64 that is available here: [continue-intellij-extension-0.0.87.zip](https://github.com/continuedev/continue/actions/runs/13192097298/artifacts/2551685736)

Before we release, it would be helpful if folks could attempt to install on their machines and see if you're able to run. Instructions on how to install from disk here: https://github.com/continuedev/continue/blob/main/extensions/intellij/CONTRIBUTING.md#installing-the-packaged-extension

In this case you'll select the downloaded ZIP rather than from `build/distributions`.

**Comment by xgdgsc** [comment]
Any preview build for vscode?

**Comment by Volko61** [comment]
> We've got a preview build for win-arm64 that is available here: [continue-intellij-extension-0.0.87.zip](https://github.com/continuedev/continue/actions/runs/13192097298/artifacts/2551685736)
> 
> Before we release, it would be helpful if folks could attempt to install on their machines and see if you're able to run. Instructions on how to install from disk here: https://github.com/continuedev/continue/blob/main/extensions/intellij/CONTRIBUTING.md#installing-the-packaged-extension
> 
> In this case you'll select the downloaded ZIP rather than from `build/distributions`.
> 
> cc @SpontaneousDuck @Volko61 @Amrsatrio @svantesorberg @xgdgsc @bold84 @mehraj43 - and others 😁 

Thanks a lot, that's so great 👍👍 
However, we mostly use it on vs code 😅
But I will install PyCharm to test it for you

**Comment by Patrick-Erichsen** [comment]
@Volko61 apologies for the delay, actually realized I needed to pull in the correct `node_sqlite3` binary for win32-arm64 as well.

Please try the VS Code release by downloading here: https://github.com/continuedev/continue/actions/runs/13233146159/artifacts/2562194902

And you can learn how to install that VSIX manually here: https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix

**Comment by Patrick-Erichsen** [comment]
I installed the new VSIX on Windows 11  running via UTM on my MBP M4 without getting any error related to LanceDB or node_sqlite3.node 👍

**Comment by Patrick-Erichsen** [comment]
@Volko61 looks like GitHub won't let me upload a VSIX in a comment so I uploaded it to Google Drive here: https://drive.google.com/file/d/13zxnVQ-Bq1f08IUpHX06Q9DS57nhowqB/view?usp=sharing

**Comment by Patrick-Erichsen** [comment]
Hi all, excited to say that this has been merged in https://github.com/continuedev/continue/pull/4019 and will be available in our next release 🚀  

Please report back here if you have issues running Continue once you've upgraded, otherwise, happy coding!

**Comment by Syktox** [comment]
Problem + 1, 
I haven't had this on my MBP M3 Pro but now I switched to my Windows 11 PC the error suddenly appeared. Any idea how I can change the needed module for the specific platform or should I just not use it and use something different.

**Comment by Patrick-Erichsen** [comment]
Closing this out, but if other folks run into the issue feel free to comment and re-open.

**Comment by Pyroboomka** [comment]
I currently try to build an extension for my use on Windows x64 PC.
I use my M1 mac, following the package process in docs:
```
sh scripts/install-all-dependecies.sh
cd extensions/vscode
npm run package
```
Grab the built .vsix file from extensions/vscode/build, install via "Install from VSIX" on windows machine, get the identical error.
Seeing that issue is marked as resolved, got any ideas what's happening here?
Using the latest main branch and latest VS Code version for windows.

Edited:
nvm, figured it out, you are supposed to `npm run package-all` to get all possible platform combinations.
